src/telemetry/telemetry.ts

```diff
@@ -0,0 +1,54 @@
+import { debugLogger } from '../utils/logging.js';
+import { AnalyticsData, AnalyticsHandler } from './sdk/analyticsHandler.sdk.js';
+import { getTelemetrySessionId, saveTelemetrySessionId } from './session.js';
+import { v4 as uuidv4 } from 'uuid';
+
+export class GenezioTelemetry {
+
+  static async getSessionId(): Promise<string> {
+    const sessionId = await getTelemetrySessionId();
+    if (!sessionId) {
+      const newSessionId = uuidv4();
+      debugLogger.debug(`[GenezioTelemetry]`, `New session id: ${newSessionId}`);
+      saveTelemetrySessionId(newSessionId);
+      return newSessionId;
+    }
+    
+    return sessionId;
+  }
+
+  public static async sendEvent(eventType: string, cloudProvider?: string, errorTrace?: string): Promise<void> {
+    if (process.env.GENEZIO_NO_TELEMETRY == "1") {
```

Comment:
We should think where we document this to let the users know that they can disabled.